packages/compiler/test/config/config.test.ts

```diff
@@ -148,3 +154,58 @@ describe("compiler: config file loading", () => {
     });
   });
 });
+
+describe("compiler: resolve compiler options", () => {
+  const tspOutputPath = normalize(`${process.cwd()}/tsp-output`);
+  describe("file discovery", () => {
+    const resolveOptions = async (path: string) => {
+      const fullPath = join(__scenarioRoot, path);
+      const [options, diagnostics] = await resolveCompilerOptions(NodeHost, {
+        entrypoint: fullPath, // not really used here
+        configPath: fullPath,
+      });
+      return { options, diagnostics };
+    };
+
+    it("resolves full path to custom config file", async () => {
+      const config = await resolveOptions("custom/myConfig.yaml");
+      config.options.config = normalize(config.options.config!);
+      config.options.outputDir = normalize(config.options.outputDir!);
+
+      deepStrictEqual(config, {
+        diagnostics: [],
+        options: {
+          config: join(__scenarioRoot, "custom/myConfig.yaml"),
+          emit: ["openapi"],
+          options: {},
+          outputDir: tspOutputPath,
+        },
+      });
+    });
+
+    it("throws error when config file not found", async () => {
+      try {
+        await resolveOptions("not-found.yaml");
+      } catch (error) {
+        return;
+      }
+      fail("should throw error");
+    });
+
+    it("resolves yaml config file", async () => {
```

Comment:
A bit confused by what this test is testing, shouldn't `--config` always only point to a config file? Do we allow it to be a directory too?

Reply:
Filed an issue to discuss this behavior https://github.com/microsoft/typespec/issues/2282. Please comment on it if it was something you expected to work and depended on.